### PR TITLE
Nifi 12505 system diagnostics

### PR DESCRIPTION
**Pull request by rfellows**
# Summary

[NIFI-12505](https://issues.apache.org/jira/browse/NIFI-12505)

* Added System Diagnostics dialog
* Refactored Status History to follow the same pattern where we use one action to initiate the load and open of the dialog rather than letting the dialog initiate the load after it is opened.

### CODE REVIEW

**Comment by mcgilman** [comment]
Will review...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/connection-status-listing/connection-status-listing.component.html [hunk 1]
```diff
@@ -41,6 +41,9 @@
                     <div>Last updated:</div>
                     <div class="refresh-timestamp">{{ loadedTimestamp$ | async }}</div>
                 </div>
+                <div>
+                    <a (click)="openSystemDiagnostics($event)">System Diagnostics</a>
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/connection-status-listing/connection-status-listing.component.html:45]: There is an explicit permission for accessing system diagnostics. In current NiFi the URL is always shown and the user gets an insufficent permissions error message when clicking it. We could consider hiding the link in this case or showing the message in a dialog. Currently, it is unhandled and nothing happens in the UI. Inspecting Dev Tools shows the `403` response.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/connection-status-listing/connection-status-listing.component.ts [hunk 2]
```diff
@@ -104,4 +108,15 @@ export class ConnectionStatusListing {
             })
         );
     }
+
+    openSystemDiagnostics(event: MouseEvent) {
+        event.stopPropagation();
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/connection-status-listing/connection-status-listing.component.ts:113]: I don't think this `stopPropagation` is needed here. It's needed in table actions which results in the user being navigated someplace. That navigation conflicted with the row selection which also changed the route. This comment applies to all instances of `openSystemDiagnostics` in all component status listings.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/system-diagnostics-dialog/system-diagnostics-dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { SystemDiagnosticsDialog } from './system-diagnostics-dialog.component';
+import { provideMockStore } from '@ngrx/store/testing';
+import { initialSystemDiagnosticsState } from '../../../state/system-diagnostics/system-diagnostics.reducer';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+describe('SystemDiagnosticsDialogComponent', () => {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/system-diagnostics-dialog/system-diagnostics-dialog.component.spec.ts:25]: This text does not match the component name. :)

**Review by mcgilman — CHANGES_REQUESTED** [review]
Thanks for the PR @rfellows! Looks great, just a few small things I noticed below.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/connection-status-listing/connection-status-listing.component.html (hunk 1 above)
Comment (rfellows) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/connection-status-listing/connection-status-listing.component.html:45]: I'll hide these links if the user does not had the appropriate permission.

**Review by mcgilman — APPROVED** [review]
Thanks for the updates @rfellows! Looks great. +1

**Pull request merged**